Notice: Fix dismiss button regression

drewblaisdell opened the pull request:
Fixes #4243

This PR updates `Notice` to separate the notice content (text, and sometimes an action) and the dismiss button, so that we can use flex boxes to arrange the latter differently than the former in small viewports.

**Testing**
*With the dismiss button (global)*
- Visit `/people/new/:site` and invite any user to your site.
- Assert that the dismiss button in the notice that appears is properly aligned in all viewports > 960px and especially < 480px:

![screen shot 2016-03-23 at 11 28 16 am](https://cloud.githubusercontent.com/assets/1130674/13996380/a98c78e8-f0ea-11e5-8a80-a42b52d65be0.png)

*With the dismiss button (inline)*
- Visit `/people/edit/:username/:site` and click on a user that is not the owner of `:site`. You may need to invite a user to `:site` if there is only one user listed.
- Update the role of this user.
- Assert that the dismiss button in the notice that appears is properly aligned in all viewports > 960px and especially < 480px:

![screen shot 2016-03-23 at 11 05 13 am](https://cloud.githubusercontent.com/assets/1130674/14001612/29030ac4-f104-11e5-9149-6a7724a770c0.png)

*With an action*
- Visit `/post/:site`.
- Publish a post.
- Assert that the notice that appears is properly aligned in all viewports > 960px and especially < 480px:

![screen shot 2016-03-23 at 11 05 22 am](https://cloud.githubusercontent.com/assets/1130674/13996647/febe4138-f0eb-11e5-98d9-5d4b39408f73.png)

- [x] Code review
- [x] Product review

Tug commented:
It seems the dismiss button is not aligned to the right in my case:

![image](https://cloud.githubusercontent.com/assets/230230/13998488/2f12e9de-f140-11e5-8792-e71b7946008c.png)

drewblaisdell commented:
Thanks for catching that, @Tug. I updated the PR to add `flex-grow: 1` to `.notice__content` to prevent that issue, and updated the testing instructions to cover the dismiss button in inline notices as well.

Tug commented:
Product :+1:

Tug commented:
Code :+1:
